packages/aws-cdk/lib/cli/cdk-toolkit.ts

```diff
@@ -1210,6 +1214,31 @@ export class CdkToolkit {
     }
   }
 
+  public async refactor(options: RefactorOptions): Promise<number> {
+    if (!options.dryRun) {
+      info('Refactor is not available yet. Too see the proposed changes, use the --dry-run flag.');
+      return 1;
+    }
+
+    if (options.selector.patterns.length > 0) {
+      warning('Refactor does not yet support stack selection. Proceeding with the default behavior (considering all stacks).');
+    }
+
+    const stacks = await this.selectStacksForList([]);
+    try {
+      const mappings = await detectRefactorMappings(stacks.stackArtifacts, this.props.sdkProvider);
```

Comment:
Hm. I thought the CLI would use the toolkit code here, but I see we haven't gotten that far yet.

Reply:
From what I understand, the idea is that both the CLI and the toolkit library share common code, but the CLI doesn't call the toolkit library. Not yet, at least.